### PR TITLE
My-Account-Country-Select

### PR DESCRIPTION
**Pull request by ainarssondors**
-Added config file for all countries
 -Added Select into My Account Address form
 -Added refmap check for Select component
 -Changed Select parameter to onGetKey
 -Added styling for My Account mobile

### CODE REVIEW

**Comment by ainarssondors** [comment]
Fixes: #119

File: src/app/component/Select/Select.style.scss [hunk 1]
```diff
@@ -92,13 +92,12 @@
             }
 
             > div > .Select-Arrow::after {
```
Comment (alfredsgenkins) [inline src/app/component/Select/Select.style.scss]: There were no any other way around such deep nesting?

File: src/app/route/MyAccountDetails/MyAccountDetails.component.js [hunk 2]
```diff
@@ -13,14 +13,17 @@ import React, { Component } from 'react';
...
 import TextPlaceholder from 'Component/TextPlaceholder';
+import CountryConfig from 'Util/Config/config';
 import { Redirect } from 'react-router';
 import './MyAccountDetails.style';
 
 const STATE_ACCOUNT_OVERVIEW = 'accountOverview';
 const STATE_EDIT_INFORMATION = 'editInformation';
 const STATE_EDIT_PASSWORD = 'editPassword';
 const STATE_UPDATE_ADDRESS = 'updateAddress';
+const DEFAULT_COUNTRY = 'LV';
```
Comment (alfredsgenkins) [inline src/app/route/MyAccountDetails/MyAccountDetails.component.js]: Let's use US

File: src/app/util/Config/config.js [hunk 3]
```diff
@@ -0,0 +1,249 @@
+export default [
```
Comment (alfredsgenkins) [inline src/app/util/Config/config.js]: Can you check if there is a region ID map? Not sure if it exists.

Comment (alfredsgenkins) [reply src/app/util/Config/config.js]: Actually, it seems Magento handles this some-how different. Please investigate. Potentially the regions could be disabled for countries, but this is not deafult values. We must do something with it...

File: src/app/route/MyAccountDetails/MyAccountDetails.style.scss [hunk 4]
```diff
@@ -120,9 +141,29 @@
...
+            }
+        }
+
+        .Select-Wrapper {
+            span {
+                top: 0;
+            }
+
+            ul {
+                max-height: 15rem;
```
Comment (alfredsgenkins) [inline src/app/route/MyAccountDetails/MyAccountDetails.style.scss]: Maybe this should be default?

File: src/app/route/MyAccountDetails/MyAccountDetails.component.js [hunk 5]
```diff
@@ -298,13 +310,17 @@ class MyAccountDetails extends Component {
...
                         />
-                        <Field
-                          type="text"
-                          label="Country"
-                          id="country_id"
-                          validation={ ['notEmpty'] }
-                          value={ country_id }
-                        />
+                        <div block="Field">
+                            <span block="MyAccountDetails" elem="Country">Country</span>
```
Comment (alfredsgenkins) [inline src/app/route/MyAccountDetails/MyAccountDetails.component.js]: Maybe we should put select in Field component? And use one interface for all inputs? I mean on Change... Not sure. Dislike labeling every select by hand.

File: src/app/util/Config/config.js (hunk 3 above)
Comment (ainarssondors) [reply src/app/util/Config/config.js]: Country list is now taken from magento. If you want to change region to select, there is the possibility to get available_regions from country. That should be a separate ticket.

File: src/app/route/MyAccountDetails/MyAccountDetails.component.js (hunk 5 above)
Comment (ainarssondors) [reply src/app/route/MyAccountDetails/MyAccountDetails.component.js]: Select is now part of Field. Can be used independently also (see: CategoryPage).

**Comment by alfredsgenkins** [comment]
@ainarssondors Is this one related to #136?

**Comment by ainarssondors** [comment]
> @ainarssondors Is this one related to #136? Should this be reviewed as well?

That's right!

**Comment by ainarssondors** [comment]
> Why does reducer is called Field.reducer and Field.dispatcher? We have already discussed that. Could you please make it Country not Field?
> 
> Also, is it able to get region_id?

File names have been changed. 
Now every country has a new parameter "available_regions" which consists of "code , id, name".
Please note that not every country has "available_regions".

File: src/app/component/Field/Field.container.js [hunk 6]
```diff
@@ -10,16 +10,16 @@
...
 
 const mapStateToProps = state => ({
-    countries: state.FieldReducer.countries
+    countries: state.CountryReducer.countries
 });
 
 const mapDispatchToProps = dispatch => ({
     getCountriesList() {
-        return FieldDispatcher.getCountriesList(dispatch);
+        return CountryDispatcher.getCountriesList(dispatch);
```
Comment (alfredsgenkins) [inline src/app/component/Field/Field.container.js]: Why is this one here? I do not want to request any countries from Field! Field is for display. Please do not connect or request, or do anything else in it except displaying specific input type! No. Please remove the logic of `getCountriesList` from Field at all. We discussed that multiple times!

Comment (ainarssondors) [reply src/app/component/Field/Field.container.js]: That was already fixed in Checkout-Country-Select.

**Comment by alfredsgenkins** [comment]
Please test everything after this merge.

File: src/app/component/Field/Field.component.js [hunk 7]
```diff
@@ -269,7 +263,7 @@ class Field extends Component {
     }
 
     renderTypeSelect() {
-        const { id, formRef, countries } = this.props;
+        const { id, formRef, countryList } = this.props;
```
Comment (alfredsgenkins) [inline src/app/component/Field/Field.component.js]: Are select only for countries? It is not, why is this still here?

**Pull request merged**